cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java

```diff
@@ -592,6 +592,19 @@ private void createCluster(VertxTestContext context, Kafka kafka, List<Secret> s
             expectedSecrets.add(EntityOperator.secretName(kafkaName));
         }
 
+        List<Secrets> caSecrets = getInitialCertificates(kafkaName);
+        if (getKafkaAssembly(kafkaName).getSpec().getClusterCa().isGenerateSecretOwnerReference()) {
+            assertThat(caSecrets.get(0).getMetadata().getOwnerReferences().size(), is(1));
+        } else {
+            assertThat(caSecrets.get(0).getMetadata().getOwnerReferences().size(), is(0));
+        }
+
+        if (getKafkaAssembly(kafkaName).getSpec().getClientsCa().isGenerateSecretOwnerReference()) {
+            assertThat(caSecrets.get(0).getMetadata().getOwnerReferences().size(), is(1));
+        } else {
+            assertThat(caSecrets.get(0).getMetadata().getOwnerReferences().size(), is(0));
+        }
+
```

Comment:
I do not think the test here really tests what it should. I do not think it triggered the code you changed in `createKafkaClusterInitialSecrets`. I think you would need to capture the secrets and check it on them. You also need to make sure that in different runs, the values are configured differently to test both situations which I don't think it happening here.